clients/drcachesim/tools/opcode_mix.cpp

```diff
@@ -255,6 +259,57 @@ cmp_val(const std::pair<int, int64_t> &l, const std::pair<int, int64_t> &r)
     return (l.second > r.second);
 }
 
+std::string
+get_category_names(uint category)
+{
+    std::string category_name;
+    if (category == DR_INSTR_CATEGORY_UNCATEGORIZED) {
+        category_name += instr_get_category_name(DR_INSTR_CATEGORY_UNCATEGORIZED);
+        return category_name;
+    }
+    if (TESTANY(DR_INSTR_CATEGORY_FP, category)) {
+        category_name += " ";
+        category_name += instr_get_category_name(DR_INSTR_CATEGORY_FP);
+    }
```

Comment:
How about a loop that marches through all the bits, to generalize this code and avoid us having to edit it when a new category is added?   Maybe don't even need a max named bit: going to the 32nd bit seems ok to future-proof.

Reply:
Whoops, this makes so much more sense. Done.